src/routes/products.js

```diff
@@ -1,69 +1,129 @@
 let router = express.Router();
-let products = require("../db/all-products.json");
 
-router.get("/*", (req, res) => {
-  if (req.url === "/") {
-    res.setHeader("Content-Type", "application/json");
-    res.send(products);
-    res.end();
-    return;
-  }
+const mongoose = require("mongoose");
+const Product = require("../../schemas/products");
 
-  if (req.url.includes("?")) {
-    let productsArr = [];
-    const productsIdsArr = req.query.ids.split(",");
+const checkProduct = productToCheck => {
+  const sku = productToCheck.sku;
+  const name = productToCheck.name;
+  const description = productToCheck.description;
+  const categories = productToCheck.categories;
+  if (
+    typeof sku === "number" &&
+    typeof name === "string" &&
+    typeof description === "string" &&
+    typeof categories === "object"
+  )
+    return true;
+  else return false;
+};
 
-    productsIdsArr.map(item => {
-      const getProductById = products.find(
-        element => element.id === Number(item)
-      );
-      if (getProductById) {
-        const productObj = {
-          id: getProductById.id,
-          sku: getProductById.sku,
-          name: getProductById.name,
-          description: getProductById.description
-        };
-        productsArr.push(productObj);
-      }
-      if (productsArr.length === 0) {
-        foundedProducts = {
-          status: "no products",
-          products: []
-        };
-      } else {
-        foundedProducts = {
-          status: "success",
-          products: productsArr
-        };
-      }
+router.get("/:productId", (req, res) => {
+  const id = req.params.productId;
+  Product.findById(id)
+    .exec()
+    .then(item => {
+      console.log(item);
+      res.status(200).json({
+        status: "success",
+        product: item
+      });
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
     });
+});
 
-    res.setHeader("Content-Type", "application/json");
-    res.send(foundedProducts);
-    res.end();
-    return;
-  }
+router.get("/", (req, res) => {
+  Product.find()
+    .exec()
+    .then(items => {
+      console.log(items);
+      res.status(200).json(items);
+    })
+    .catch(err => {
+      console.log(err);
+      res.status(500).json({ error: err });
+    });
+});
 
-  let oneProductId = req.url.slice(req.url.lastIndexOf("/") + 1);
-  const getOneProductById = products.find(
-    item => item.id.toString() === oneProductId
-  );
-  if (getOneProductById) {
-    res.setHeader("Content-Type", "application/json");
-    res.send(getOneProductById);
-    res.end();
-  } else {
-    res.setHeader("Content-Type", "application/json");
-    res.send(
-      JSON.stringify({
-        status: "no products",
-        products: []
+router.post("/", jsonParser, (req, res) => {
+  if (checkProduct(req.body)) {
+    console.log("Validation complete!");
+    const product = new Product({
+      _id: new mongoose.Types.ObjectId(),
```

Comment:
Kindly ask you to fix all the issues that I've mentioned above not only in the marked spot, but in other similar places. It's hard to put everywhere comment